ui/src/app/groupings/[groupingPath]/_components/export-dropdown.tsx

```diff
@@ -1,16 +1,129 @@
 'use client';
 
+import { useState, useRef } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+    faFileExport,
+    faCaretDown,
+    faUsers,
+    faUserPlus,
+    faUserMinus,
+    faIdCard
+} from '@fortawesome/free-solid-svg-icons';
+import { Tooltip, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip';
+import { getGroupingMembers } from '@/lib/actions';
 import { Button } from '@/components/ui/button';
-import { FileInput, ChevronDown } from 'lucide-react';
 
-const ExportDropdown = () => {
+interface Member {
+    lastName: string;
+    firstName: string;
+    uid: string;
+    uhUuid: string;
+}
+
+interface GroupData {
+    members: Member[];
+}
+
+const ExportDropdown = ({ groupPath }: { groupPath: string }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    const memberTypes = [
+        { type: '', label: 'All Members', icon: faUsers },
+        { type: 'basis', label: 'Basis', icon: faIdCard },
+        { type: 'include', label: 'Include', icon: faUserPlus },
+        { type: 'exclude', label: 'Exclude', icon: faUserMinus }
+    ];
+
+    const fetchGroupData = async (memberType: string) => {
+        const fullPath = memberType === '' ? groupPath : `${groupPath}:${memberType}`;
+        const params = { page: undefined, size: undefined, sortString: 'name', isAscending: true };
+        const data: GroupData = await getGroupingMembers(fullPath, params);
+        exportFile(data, memberType);
+    };
+
+    const exportFile = (data: GroupData, memberType: string) => {
+        const members = data?.members;
+        const groupName = groupPath.split(':').pop();
+        const headers = ['Last', 'First', 'Username', 'UH Number', 'Email'];
+
+        const rows = members.map((member: Member) => {
```

Comment:
The two interfaces can be removed and GroupData can be removed from line 42. On line 46, replace GroupData type with GroupingMembers. This way, we use the type that is already defined when calling `await getGroupingMembers(fullPath, params)`.